src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -241,29 +259,79 @@ public IImageInfo Identify(BufferedReadStream stream, CancellationToken cancella
                         {
                             case PngChunkType.Header:
                                 this.ReadHeaderChunk(pngMetadata, chunk.Data.GetSpan());
+                                this.isHeaderChunkReached = true;
                                 break;
                             case PngChunkType.Physical:
+                                if (this.colorMetadataOnly)
+                                {
+                                    this.SkipChunkDataAndCrc(chunk);
+                                    break;
+                                }
+
                                 this.ReadPhysicalChunk(metadata, chunk.Data.GetSpan());
                                 break;
                             case PngChunkType.Gamma:
+                                if (this.colorMetadataOnly)
+                                {
+                                    this.SkipChunkDataAndCrc(chunk);
+                                    break;
+                                }
+
                                 this.ReadGammaChunk(pngMetadata, chunk.Data.GetSpan());
                                 break;
                             case PngChunkType.Data:
                                 this.SkipChunkDataAndCrc(chunk);
+                                break;
+                            case PngChunkType.Transparency:
+                                byte[] alpha = new byte[chunk.Length];
+                                chunk.Data.GetSpan().CopyTo(alpha);
+                                this.paletteAlpha = alpha;
+                                this.AssignTransparentMarkers(alpha, pngMetadata);
+
+                                if (this.colorMetadataOnly && this.isHeaderChunkReached)
```

Comment:
According to the spec, IDAT is always the first, so - in theory - we don't need  to track if we reached it. I wonder if there are malformed PNG-s in the wild which do not follow this rule.

@brianpopow any thoughts?

Reply:
You mean `IHDR` is always first I guess? I will go through my collection of "weird and broken images" to check if i can find any which has not `IHDR` first.

Reply:
> You mean IHDR is always first I guess? 

Yes, sorry for the typo.

Reply:
I could not find any png which do not start with `IHDR`. Nonetheless I am still in favor of keeping it as it is, just to be sure,  because in theory such png can be created.